crates/typed-store/src/rocks/mod.rs

```diff
@@ -679,24 +663,25 @@ impl RocksDBBatch {
 
 #[derive(Debug, Default)]
 pub struct MetricConf {
-    pub db_name_override: Option<String>,
+    pub db_name: String,
     pub read_sample_interval: SamplingInterval,
     pub write_sample_interval: SamplingInterval,
     pub iter_sample_interval: SamplingInterval,
 }
 
 impl MetricConf {
-    pub fn with_db_name(db_name: &str) -> Self {
```

Comment:
Could we maybe add an `assert` to ensure that a non empty string db name is passed?